code/scripting/api/libs/ui.cpp

```diff
@@ -3200,135 +3283,255 @@ ADE_FUNC(setGameType,
+	nullptr,
+	"Cancels Multi Client Setup and leaves the game.",
+	nullptr,
+	nullptr)
 {
+	SCP_UNUSED(L);
 
-	bool observer = false;
-	ade_get_args(L, "*|b", &observer);
+	multi_quit_game(PROMPT_CLIENT);
 
-	if (Active_games.empty()) {
-		multi_common_add_notify(XSTR("No games found!", 757));
-		return ADE_RETURN_FALSE;
-	} else if (Multi_join_selected_item == nullptr) {
-		multi_common_add_notify(XSTR("No game selected!", 758));
-		return ADE_RETURN_FALSE;
-	} else if (Multi_join_sent_stamp.isValid() && !ui_timestamp_elapsed(Multi_join_sent_stamp)) {
-		multi_common_add_notify(XSTR("Still waiting on previous join request!", 759));
-	} else {
-		// otherwise, if he's already played PXO games, warn him
+	return ADE_RETURN_NIL;
+}
 
-		if (Player->flags & PLAYER_FLAGS_HAS_PLAYED_PXO) {
-			if (!multi_join_warn_pxo()) {
-				return ADE_RETURN_FALSE;
-			}
-		}
+ADE_FUNC(runNetwork,
+	l_UserInterface_MultiClientSetup,
+	nullptr,
+	"Runs the network required commands to update the lists and run the chat",
+	nullptr,
+	nullptr)
+{
+	SCP_UNUSED(L);
 
-		// send a join request packet
-		multi_join_send_join_request(observer);
-		return ADE_RETURN_TRUE;
-	}
+	multi_game_client_setup_do_frame(true);
 
-	return ADE_RETURN_FALSE;
+	return ADE_RETURN_NIL;
 }
 
-ADE_VIRTVAR(StatusText, l_UserInterface_MultiJoinGame, nullptr, "The current status text", "string", "the status text")
+//**********SUBLIBRARY: UserInterface/MultiSync
+ADE_LIB_DERIV(l_UserInterface_MultiSync,
+	"MultiSync",
+	nullptr,
+	"API for accessing data related to the Multi Sync UI.",
+	l_UserInterface);
+
+ADE_FUNC(initMultiSync,
+	l_UserInterface_MultiSync,
+	nullptr,
+	"Makes sure everything is done correctly to begin the multi sync ui.",
+	nullptr,
+	nullptr)
 {
-	if (ADE_SETTING_VAR) {
-		LuaError(L, "This property is read only.");
-	}
+	SCP_UNUSED(L);
 
-	return ade_set_args(L, "s", Multi_common_notify_text);
+	multi_sync_init(true);
+
+	return ADE_RETURN_NIL;
 }
 
-ADE_VIRTVAR(InfoText, l_UserInterface_MultiJoinGame, nullptr, "The current info text", "string", "the info text")
+ADE_FUNC(closeMultiSync,
+	l_UserInterface_MultiSync, "boolean QuitGame",
+	"Closes MultiSync. If QuitGame is true then it cancels and leaves the game automatically.",
+	nullptr,
+	nullptr)
 {
-	if (ADE_SETTING_VAR) {
-		LuaError(L, "This property is read only.");
+	bool choice = true;
+	if (!ade_get_args(L, "b", &choice))
+		return ADE_RETURN_NIL;
+
+	if (!choice) {
+		multi_sync_close(true);
+	} else {
+		multi_quit_game(PROMPT_CLIENT);
 	}
 
-	return ade_set_args(L, "s", Multi_common_all_text);
+	return ADE_RETURN_NIL;
 }
 
-ADE_LIB_DERIV(l_Active_Games, "ActiveGames", nullptr, nullptr, l_UserInterface_MultiJoinGame);
-ADE_INDEXER(l_Active_Games,
-	"number Index",
-	"Array of active games",
-	"active_game",
-	"active game handle, or invalid handle if index is invalid")
+ADE_FUNC(runNetwork,
+	l_UserInterface_MultiSync,
+	nullptr,
+	"Runs the network required commands to update the lists and run the chat",
```

Comment:
This only covers multi sync, correct? So not list updates and chat.

Reply:
Yup. Good catch.